### ISSUE TITLE
[Bug]: media: hover pointer: fine style not applied on firefox

### ISSUE THREAD

**Issue by axyz** [microsoft/playwright#7769]
### Playwright version

1.12.3

### Operating system

Linux

### What browsers are you seeing the problem on?

Firefox

### Other information

node 16

### What happened? / Describe the bug

I'm using this media query to wrapp all the hover specific classes:

```
@media (hover: hover) and (pointer: fine) {
  .box:hover {
    background-color: green;
  }
}
```

Using this pen as a reference: https://codepen.io/axyz/pen/RwVLbRM

It shows a simple red box. When hovered on a pointer device like a regular desktop it should become green. It works both on chrome and firefox.

When taking a screenshot with playwright though, after simulating a hover event, the box is correctly green on chrome, but it is red on firefox.

It looks like the media expression is not matched in headless mode, but this is just my guess.

### Code snippet to reproduce your bug

```shell
(async () => { 
  const browser = await require("playwright").firefox.launch(); // changing this to chromium outputs the correct png with a green box.
  const page = await browser.newPage(); 
  await page.goto("https://cdpn.io/axyz/fullpage/RwVLbRM"); 
  const iframe = await page.frames()[1]; 
  const box = await iframe.$(".box"); 
  await box.hover(); 
  await page.screenshot({path: "test.png"}); 
  await browser.close() 
})()
```

**Comment by pavelfeldman** [comment]
I can reproduce this on Linux, works fine on macOS.

**Comment by aslushnikov** [comment]
### TL;DR: it's a firefox bug, but you can workaround using Firefox Prefs.

@axyz This happens due to Firefox Headless on GTK forces pointer capabilities to "none": 

https://github.com/mozilla/gecko-dev/blob/df6434d2ebfdf2b5f89f205fc81d60d64a774fe1/widget/headless/HeadlessLookAndFeelGTK.cpp#L249-L255

I can reproduce on current Firefox Stable M94 with the following line:

```bash
firefox --headless --screenshot shot.png file:///root/index.html
```

 where `index.html` is:

```html
<style>
@media (pointer: fine) {
  html { background-color: green !important; }
}
html { background-color: red; }
</style>
```

So it's better to file this upstream to firefox repo. 

Meanwhile, you can actually have it working smoothly by overriding a few firefoxUserPrefs; here's what worked for me:

```js
(async () => {
  const browser = await require("@playwright/test").firefox.launch({
    firefoxUserPrefs: {
      "ui.primaryPointerCapabilities": 0x02 | 0x04,
      "ui.allPointerCapabilities": 0x02 | 0x04,
    },
  });
  const page = await browser.newPage();
  await page.setContent(`
    <style>
      @media (pointer: fine) {
        html { background-color: green !important; }
      }
      html { background-color: red; }
    </style>
  `);
  await page.screenshot({ path: "test.png" });
  await browser.close()
})();
```

Hope this helps!

**Comment by LukasTy** [comment]
Hello.
Been experiencing the same issue with headless chrome and finally found the solution after scouring their flag documentation. Do note that chrome only supports `flags` for configuration and not `prefs`.

For anyone also in need to force custom pointer support:
```
'--blink-settings=primaryPointerType=4'
```
Sources:
https://source.chromium.org/chromium/chromium/src/+/main:ui/base/pointer/pointer_device.h;l=46
https://chromium.googlesource.com/chromium/src/third_party/+/master/blink/renderer/core/frame/settings.json5#647

**Comment by jwir3** [comment]
Does anyone happen to know if this was filed as a bugzilla ticket (i.e. on Firefox upstream?)

**Comment by bodograumann** [comment]
You can define the workaround in your playwright config:

```javascript
{
  use: {
    launchOptions: {
      firefoxUserPrefs: {
        "ui.primaryPointerCapabilities": 0x02 | 0x04,
        "ui.allPointerCapabilities": 0x02 | 0x04,
      },
    },
  },
}
```
